src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendVideoBookingService.kt

```diff
@@ -0,0 +1,96 @@
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.LocationValidator
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonersearch.PrisonerValidator
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.VideoBooking
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.Prisoner
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AmendVideoBookingRequest
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.BookingType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.CourtRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonAppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.ProbationTeamRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import java.time.LocalDateTime
+
+@Service
+class AmendVideoBookingService(
+  private val courtRepository: CourtRepository,
+  private val probationTeamRepository: ProbationTeamRepository,
+  private val videoBookingRepository: VideoBookingRepository,
+  prisonAppointmentRepository: PrisonAppointmentRepository,
+  prisonRepository: PrisonRepository,
+  locationValidator: LocationValidator,
+  prisonerValidator: PrisonerValidator,
+) : CreateVideoAppointmentService(prisonRepository, prisonAppointmentRepository, locationValidator, prisonerValidator) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  fun amend(videoBookingId: Long, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val booking = videoBookingRepository.findById(videoBookingId)
+      .orElseThrow { EntityNotFoundException("Video booking with ID $videoBookingId not found.") }
+      .also {
+        require(BookingType.valueOf(it.bookingType) == request.bookingType) {
+          "The booking type ${it.bookingType} does not match the requested type ${request.bookingType}."
+        }
+      }
+
+    return when (BookingType.valueOf(booking.bookingType)) {
+      BookingType.COURT -> amendCourt(booking, request, amendedBy)
+      BookingType.PROBATION -> amendProbation(booking, request, amendedBy)
+    }
+  }
+
+  private fun amendCourt(booking: VideoBooking, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val court = courtRepository.findByCode(request.courtCode!!)
+      ?.also { require(it.enabled) { "Court with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Court with code ${request.courtCode} not found")
+
+    val prisoner = request.prisoner().validate()
+
+    booking.apply {
```

Comment:
Not essential but perhaps an amend method on the entity which takes in what is needed?

This would avoid the need to supply and amended time and allow validation to be added to the entity e.g. is the court enabled? Can you set court? It could be a probation booking by mistake ...

The same would go for amend a probation booking.